tests/test_export.py

```diff
@@ -443,6 +443,62 @@ def test_overlap_end_code():
     validate_code(cells[:4], cells[4:], cells)
 
 
+def test_handle_cell_magic_before_start():
```

Comment:
IIUC all of the tests you added are actually testing `handle_line_magic` and not `handle_cell_magic`:
1. Please name correctly
2. please add tests for both things